modules/auxiliary/gather/shodan_honeyscore.rb

```diff
@@ -42,6 +42,11 @@ def print_score(score)
   end
 
   def run
+    # check our API key is somewhat sane
+    unless /^[a-z\d]{32}$/i.match?(datastore['SHODAN_APIKEY'])
+      fail_with(Failure::BadConfig, 'Shodan API key should be 32 characters a-z,A-Z,0-9.')
```

Comment:
Any thoughts on dumping out the unexpected character? It might help with debugging 👀

Reply:
I didn't want to overcomplicate a copy/paste operation. Most likely if this triggered it would be a length issue is what I'm guessing